examples/trading-runner/src/uniswap/constants.ts

```diff
@@ -0,0 +1,124 @@
+// This file stores web3 related constants such as addresses, token definitions, ETH currency references and ABIs
+
+import { FeeAmount } from "@uniswap/v3-sdk";
+import { SupportedChainId, Token, Percent } from "@uniswap/sdk-core";
+import uniV3UniversalRouterContractABI from "./abi/univ3-universal-router-contract-abi.json";
+import wethContractABI from "./abi/weth-contract-abi.json";
+
+// Environment
+
+export enum Environment {
+  GOERLI = "goerli",
+  MAINNET = "mainnet",
+}
+
+// Transaction-related configs
+
+export const DEFAULT_MAX_FEE_PER_GAS = 10000000000; // 10 gwei
+export const DEFAULT_MAX_PRIORITY_FEE_PER_GAS = 10000000000; // 10 gwei
+export const DEFAULT_TOKEN_APPROVAL_AMOUNT = 1000; // whole units
+export const DEFAULT_SLIPPAGE_TOLERANCE = new Percent(50, 10_000); // 50 bips, or 0.50%
+export const FEE_AMOUNT = FeeAmount.MEDIUM;
+export const NATIVE_TRANSFER_GAS_LIMIT = 21000;
+export const ERC20_TRANSFER_GAS_LIMIT = 200000; // conservatively high
+export const GAS_MULTIPLIER = 2;
+
+// Contract Addresses
+
+export const POOL_FACTORY_CONTRACT_ADDRESS =
+  "0x1F98431c8aD98523631AE4a59f267346ea31F984"; // mainnet + goerli
+export const QUOTER_CONTRACT_ADDRESS =
+  "0x61fFE014bA17989E743c5F6cB21bF9697530B21e"; // mainnet + goerli
+export const SWAP_ROUTER_ADDRESS = "0xE592427A0AEce92De3Edee1F18E0157C05861564"; // mainnet + goerli
+export const UNI_V3_UNIVERSAL_ROUTER_CONTRACT_ADDRESS_GOERLI =
+  "0x4648a43b2c14da09fdf82b161150d3f634f40491";
+
+// Contract call function signatures. Must be lowercase.
+
+export const APPROVE_SIGNATURE = "0x095ea7b3";
```

Comment:
these are function selectors. [ref](https://ethereum.stackexchange.com/questions/135205/what-is-a-function-signature-and-function-selector-in-solidity-and-evm-language)

Reply:
🎯 spot on. ty!